src/examples/zkapps/voting/test.ts

```diff
@@ -28,46 +41,769 @@ export async function testSet(
+
+      invalidSet.voting.voterRegistration(m);
+
+      invalidSet.voting.sign(votingKey);
     });
 
-    voting.deploy({ zkappKey: params.votingKey });
-    candidateContract.deploy({ zkappKey: params.candidateKey });
-    voterContract.deploy({ zkappKey: params.voterKey });
+    tx.send();
+  } catch (err: any) {
+    if (!err.toString().includes('precondition_unsatisfied')) {
+      throw Error('Transaction should have failed but went through!');
+    }
+  }
+
+  const initialRoot = votersStore.getRoot();
+
+  console.log('deploying testing phase 2 contracts');
+
+  let sequenceOverflowSet = await deployContracts(
+    contracts,
+    params,
+    votersStore.getRoot(),
+    candidatesStore.getRoot(),
+    votesStore.getRoot()
+  );
 
-    // setting the merkle root
-    voterContract.committedMembers.set(storage.votersStore.getRoot());
-    candidateContract.committedMembers.set(storage.candidatesStore.getRoot());
-    voting.committedVotes.set(storage.votesStore.getRoot());
+  /*
+    test case description:
+      overflowing maximum amount of sequence events allowed in the reducer (32 default)
+    
+    preconditions:
+      - x
 
-    // setting the initial sequence events hash
-    voterContract.accumulatedMembers.set(
-      Experimental.Reducer.initialActionsHash
+    tested cases:
+      - emitted 33 sequence events and trying to reduce them
+
+    expected results:
+      - throws an error
+
+  */
+
+  console.log('trying to overflow sequence events (default 32)');
+
+  console.log(
+    'emitting more than 32 sequence events without periodically updating them'
+  );
+  for (let index = 0; index <= 32; index++) {
+    try {
+      let tx = await Mina.transaction(sequenceOverflowSet.feePayer, () => {
+        let m = Member.from(
+          PrivateKey.random().toPublicKey(),
+          Field.zero,
+          UInt64.from(15)
+        );
+        sequenceOverflowSet.Local.addAccount(m.publicKey, m.balance.toString());
+
+        sequenceOverflowSet.voting.voterRegistration(m);
+
+        sequenceOverflowSet.voting.sign(votingKey);
+      });
+      tx.send();
+    } catch (error) {
+      throw new Error('Transaction failed!');
+    }
+  }
+
+  if (sequenceOverflowSet.voterContract.reducer.getActions({}).length < 32) {
+    throw Error(
+      `Did not emitted expected sequence events! Only emitted ${
+        sequenceOverflowSet.voterContract.reducer.getActions({}).length
+      }`
     );
-    candidateContract.accumulatedMembers.set(
-      Experimental.Reducer.initialActionsHash
+  }
+
+  try {
+    let tx = await Mina.transaction(sequenceOverflowSet.feePayer, () => {
+      sequenceOverflowSet.voting.authorizeRegistrations();
+      sequenceOverflowSet.voting.sign(votingKey);
+    });
+    tx.send();
+  } catch (err: any) {
+    if (!err.toString().includes('the maximum number of lists of actions')) {
+      throw Error('Transaction should have failed but went through!');
+    }
```

Comment:
Same comment as above about handling the cases when the transaction fails for any other reason than the expected failure case. 

    ```
if (!err.toString().includes('the maximum number of lists of actions')) {
      throw Error('Transaction should have failed but went through!');
    } else {
      // Throw an error on all unexpected failure cases
      throw Error(error)
  }
```